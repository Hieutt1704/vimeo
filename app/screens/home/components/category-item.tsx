import React from "react"
import { View } from "react-native"
import { AutoImage, Text } from "../../../components"
import { Category } from "../../../models"
import homeScreenStyles from "../home-screen.styles"

const BACKGROUND_IMAGE =
  "https://i.vimeocdn.com/video/1486234694-2789256b3e16a01f78abaebcc3eeb1ef872b63bc8bfda834c8c4cc6b11fb578b-d"

interface CategoryItemProps {
  testID?: string
  category: Category
}

export const CategoryItem = ({ category, testID }: CategoryItemProps) => {
  return (
    <View testID={testID} style={homeScreenStyles.categoryContainer}>
      <AutoImage source={{ uri: BACKGROUND_IMAGE }} style={homeScreenStyles.categoryBackground} />
      <View style={homeScreenStyles.categoryInfo}>
        <AutoImage source={{ uri: category.image }} style={homeScreenStyles.categoryIcon} />
        <Text style={homeScreenStyles.categoryName}>{category.name}</Text>
      </View>
    </View>
  )
}
