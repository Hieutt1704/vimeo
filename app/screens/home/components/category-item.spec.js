import { render, screen } from "@testing-library/react-native"
import React from "react"
import { CategoryItem } from "./category-item"

test("match snapshot", () => {
  render(
    <CategoryItem
      category={{
        key: "abcdef",
        id: "favourite",
        name: "favourite",
        lastUpdate: "022-08-20T18:51:49+00:00",
        image: "",
      }}
    />,
  )
  expect(screen.toJSON()).toMatchSnapshot()
})
