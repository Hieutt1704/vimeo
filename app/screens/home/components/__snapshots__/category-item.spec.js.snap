// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<View
  style={
    Object {
      "marginLeft": 8,
    }
  }
>
  <RCTImageView
    internal_analyticTag={null}
    resizeMode="cover"
    source={
      Array [
        Object {
          "uri": "https://i.vimeocdn.com/video/1486234694-2789256b3e16a01f78abaebcc3eeb1ef872b63bc8bfda834c8c4cc6b11fb578b-d",
        },
      ]
    }
    style={
      Object {
        "height": 363,
        "overflow": "hidden",
        "width": 363,
      }
    }
  />
  <View
    style={
      Object {
        "height": 24,
        "tintColor": "#ffffff",
        "width": 24,
      }
    }
  >
    <RCTImageView
      internal_analyticTag={null}
      resizeMode="cover"
      source={
        Array [
          Object {
            "uri": "",
          },
        ]
      }
      style={
        Object {
          "height": 24,
          "overflow": "hidden",
          "tintColor": "#ffffff",
          "width": 24,
        }
      }
      tintColor="#ffffff"
    />
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Helvetica",
            "fontSize": 15,
          },
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 8,
          },
        ]
      }
    >
      favourite
    </Text>
  </View>
</View>
`;
